src/Accounts/Accounts/CommonModule/ContextAdapter.cs

```diff
@@ -191,6 +192,35 @@ public object GetParameterValue(string resourceId, string moduleName, Invocation
             return string.Empty;
         }
 
+        internal async Task<HttpResponseMessage> AuthenticationHelper(IAzureContext context, string endpointResourceIdKey, string endpointSuffixKey, HttpRequestMessage request, CancellationToken cancelToken, Action cancelAction, SignalDelegate signal, NextDelegate next, TokenAudienceConverterDelegate tokenAudienceConverter = null)
+        {
+            IAccessToken accessToken = await AuthorizeRequest(context, request, cancelToken, endpointResourceIdKey, endpointSuffixKey, tokenAudienceConverter);
+            var response = await next(request, cancelToken, cancelAction, signal);
+
+            if (response.MatchClaimsChallengePattern())
+            {
+                //get token again with claims challenge
+                if (accessToken is IClaimsChallengeProcessor processor)
+                {
+                    try
+                    {
+                        var claimsChallenge = ClaimsChallengeUtilities.GetClaimsChallenge(response);
+                        if (!string.IsNullOrEmpty(claimsChallenge))
+                        {
+                            await processor.OnClaimsChallenageAsync(request, claimsChallenge, cancelToken).ConfigureAwait(false);
+                            request = request.CloneAndDispose(request.RequestUri, request.Method);
```

Comment:
1. Cloneanddispose will not clone the request content. 
2. The clone should be moved to line#198, which is before sending the request.